src/google/cloud/ndb/model.py

```diff
@@ -3354,10 +3357,184 @@ def _now():
 
 
 class StructuredProperty(Property):
-    __slots__ = ()
+    """A Property whose value is itself an entity.
 
-    def __init__(self, *args, **kwargs):
-        raise NotImplementedError
+    The values of the sub-entity are indexed and can be queried.
+    """
+
+    _modelclass = None
+
+    def __init__(self, modelclass, name=None, **kwargs):
+        super(StructuredProperty, self).__init__(name=name, **kwargs)
+        if self._repeated:
+            if modelclass._has_repeated:
+                raise TypeError(
+                    "This StructuredProperty cannot use repeated=True "
+                    "because its model class (%s) contains repeated "
+                    "properties (directly or indirectly)."
+                    % modelclass.__name__
+                )
+        self._modelclass = modelclass
+
+    def _get_value(self, entity):
+        """Override _get_value() to *not* raise UnprojectedPropertyError."""
```

Comment:
Could you expand this docstring to say something about why this intentionally raises unprojectedpropertyerror and under what circumstances you'd want to override that behavior?